library/init/category/except.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jared Roesch, Sebastian Ullrich
+
+The except monad transformer.
+-/
+
+prelude
+import init.category.alternative init.category.lift
+universes u v w
+
+inductive except (ε : Type u) (α : Type v)
+| error {} : ε → except
+| ok {} : α → except
+
+class monad_except (ε : out_param (Type u)) (m : Type v → Type w) :=
```

Comment:
I'm assuming the purpose of `monad_except` is to define a generic `throw` and `catch`.
This is useful because the interface is implemented by `except_t` and `io`.
Am I correct? If yes, we should add a comment motivating this type class.
Note that, we don't have `monad_state`. That is, `get` and `put` are tied to `state_t`.